bcda/models/models.go

```diff
@@ -332,10 +334,52 @@ type CCLFFile struct {
 // https://www.cms.gov/Medicare/New-Medicare-Card/Understanding-the-MBI-with-Format.pdf
 type CCLFBeneficiary struct {
 	gorm.Model
-	FileID        uint   `gorm:"not null"`
-	HICN          string `gorm:"type:varchar(11);not null"`
-	MBI           string `gorm:"type:char(11);not null"`
-	BeneficiaryID uint
+	CCLFFile     CCLFFile
+	FileID       uint   `gorm:"not null"`
+	HICN         string `gorm:"type:varchar(11);not null"`
+	MBI          string `gorm:"type:char(11);not null"`
+	BlueButtonID string `gorm:"type: text"`
+}
+
+// This method will ensure that a valid BlueButton ID is returned.
+// If you use cclfBeneficiary.BlueButtonID you will not be guaranteed a valid value
+func (cclfBeneficiary *CCLFBeneficiary) GetBlueButtonID(bb client.APIClient) (blueButtonID string, err error) {
+	// If this is set already, just give it back.
+	if cclfBeneficiary.BlueButtonID != "" {
+		return cclfBeneficiary.BlueButtonID, nil
+	}
+	var oldRecord CCLFBeneficiary
+	db := database.GetGORMDbConnection()
+	defer db.Close()
+
+	// find another record with this HICN.  If it has a value use it.
+	// Find the first record with a matching HICN and an non empty BlueButtonID
+	db.Where("hicn = ? and blue_button_id <> ''", cclfBeneficiary.HICN).First(&oldRecord)
+	if oldRecord.BlueButtonID != "" {
+		return oldRecord.BlueButtonID, nil
+	}
+
+	// didn't find a local value, need to ask BlueButton
+	jsonData, err := bb.GetBlueButtonIdentifier(client.HashHICN(cclfBeneficiary.HICN))
```

Comment:
I might be missing it, but shouldn't we be updating the DB with the info so it optimizes the request the next time?

Reply:
Thats fair.  Just wanted to make sure we weren't forgetting to do that somewhere.  Do we need a separate function to do that work?  Do you think its something that is out of scope of this ticket?  If so, we'll need a follow up ticket.

Reply:
I was planning on doing it in 1233 when I started actually getting the values.

Reply:
I updated BCDA-1233 with a link to this convo, just so it doesn't get lost.